contracts/BaseStrategy.sol

```diff
@@ -306,85 +266,55 @@ abstract contract BaseStrategy {
-     *  contract is deployed.
-     * @dev `_vault` should implement `VaultAPI`.
-     * @param _vault The address of the Vault responsible for this Strategy.
-     * @param _strategist The address to assign as `strategist`.
-     * The strategist is able to change the reward address
-     * @param _rewards  The address to use for pulling rewards.
-     * @param _keeper The adddress of the _keeper. _keeper
-     * can harvest and tend a strategy.
-     */
     function _initialize(
         address _vault,
         address _strategist,
         address _rewards,
-        address _keeper
+        address _keeper,
+        uint256 _changeLimitRatio
     ) internal {
         require(address(want) == address(0), "Strategy already initialized");
 
         vault = VaultAPI(_vault);
         want = IERC20(vault.token());
-        want.safeApprove(_vault, uint256(-1)); // Give Vault unlimited access (might save gas)
+        want.safeApprove(_vault, uint256(-1));
         strategist = _strategist;
         rewards = _rewards;
         keeper = _keeper;
+        changeLimitRatio = changeLimitRatio;
 
-        // initialize variables
         minReportDelay = 0;
         maxReportDelay = 86400;
         profitFactor = 100;
         debtThreshold = 0;
 
-        vault.approve(rewards, uint256(-1)); // Allow rewards to be pulled
+        vault.approve(rewards, uint256(-1));
     }
 
-    /**
-     * @notice
-     *  Used to change `strategist`.
-     *
-     *  This may only be called by governance or the existing strategist.
-     * @param _strategist The new address to assign as `strategist`.
-     */
     function setStrategist(address _strategist) external onlyAuthorized {
         require(_strategist != address(0));
         strategist = _strategist;
         emit UpdatedStrategist(_strategist);
     }
 
-    /**
-     * @notice
-     *  Used to change `keeper`.
-     *
-     *  `keeper` is the only address that may call `tend()` or `harvest()`,
-     *  other than `governance()` or `strategist`. However, unlike
-     *  `governance()` or `strategist`, `keeper` may *only* call `tend()`
-     *  and `harvest()`, and no other authorized functions, following the
-     *  principle of least privilege.
-     *
-     *  This may only be called by governance or the strategist.
-     * @param _keeper The new address to assign as `keeper`.
-     */
     function setKeeper(address _keeper) external onlyAuthorized {
         require(_keeper != address(0));
         keeper = _keeper;
         emit UpdatedKeeper(_keeper);
     }
 
-    /**
-     * @notice
-     *  Used to change `rewards`. EOA or smart contract which has the permission
-     *  to pull rewards from the vault.
-     *
-     *  This may only be called by the strategist.
-     * @param _rewards The address to use for pulling rewards.
-     */
+    function setStrategyEnforeChangeLimit(bool _enforeChangeLimit) external onlyAuthorized {
```

Comment:
Can we create a new modifier that contains these roles governance or vault.management? since these two are msigs i believe this exceptional case should be hands on deck to every one of accepting a big price per share change.

Maybe add modifier to be named `onlyGovOrManagement`